service/storage_proxy.cc

```diff
@@ -107,6 +107,12 @@
 #include "mutation_partition_view.hh"
 #include "service/paxos/paxos_state.hh"
 #include "gms/feature_service.hh"
+#include "cql3/selection/selection.hh"
+#include "cql3/cql_config.hh"
+#include "cql3/query_options.hh"
+#include "cql_serialization_format.hh"
+#include "service/pager/query_pagers.hh"
+#include "cql3/result_set.hh"
```

Comment:
I think this should be a new service. As written, we create circular dependency between storage_proxy and cql3. We're also running the risk of a deadlock, if both super-coordinator and coordinator requests take resources from the same pool.